submissions/aymkin/OOP-JS (tjsw)/index.js

```diff
@@ -0,0 +1,80 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+
+Code repository: https://github.com/aymkin/a-tiny-JS-world/tree/tiny-js-world-oop-aymkin
+Web app: https://aymkin.github.io/a-tiny-JS-world/
+   */
+
+function Inhobitant(species, name, gender, friends, say) {
+	this.species = species;
+	this.name = name;
+	this.gender = gender;
+	this.friends = friends;
+	this.say = say;
+}
+
+Inhobitant.prototype.makeSound = function() {
+	return this.say;
+};
+
+Inhobitant.prototype.getFriend = function() {
+	return this.hasOwnProperty('friends')
+		? `I have friends: ${this.friends.join(', ')}.`
+		: `I have no friends`;
+};
+
+Inhobitant.prototype.getHandLegs = function() {
```

Comment:
Class `Inhobitant` doesn't implement `.legs` and `.hands`. It is counter-SOLID to implement any logic based on properties class doesn't own.

Imagine, you will have to implement more species like rhinos (and tell a number of horns) or birds (and tell a number of wings) or make the hierarchy of classes deeper. Handling of specific properties should be done at an appropriate level of class hierarchy.

Please, fix.

Reply:
done